llpc/lower/llpcSpirvLowerGlobal.cpp

```diff
@@ -263,86 +263,95 @@ bool SpirvLowerGlobal::runImpl(Module &module) {
 }
 
 // =====================================================================================================================
-// Visits "return" instruction.
-//
-// @param retInst : "Ret" instruction
-void SpirvLowerGlobal::visitReturnInst(ReturnInst &retInst) {
-  // Skip if "return" instructions are not expected to be handled.
-  if (!static_cast<bool>(m_instVisitFlags.checkReturn))
-    return;
-
-  // We only handle the "return" in entry point
-  if (retInst.getParent()->getParent()->getLinkage() == GlobalValue::InternalLinkage)
-    return;
-
-  assert(m_retBlock); // Must have been created
-  BranchInst::Create(m_retBlock, retInst.getParent());
-  m_retInsts.insert(&retInst);
+// Handle "return" instructions.
+void SpirvLowerGlobal::handleReturnInst() {
+  for (Function &function : m_module->functions()) {
+    for (BasicBlock &block : function) {
+      Instruction *terminator = block.getTerminator();
+      if (!terminator || terminator->getOpcode() != Instruction::Ret)
+        continue;
+      // We only handle the "return" in entry point
+      if (terminator->getParent()->getParent()->getLinkage() == GlobalValue::InternalLinkage)
```

Comment:
`function->getLinkage()` is shorter ;)
Also, this can be tested before the basic block loop.